### PR TITLE
fix(cli): auto-detect the TLS version based on current node version

### PR DESCRIPTION
**Pull request by byCedric**
# Why

After debugging this issue https://github.com/byCedric/expo-monorepo-example/issues/84, it seems that we are having trouble with the internals of `expo run:ios`.

What's happening here are two major things:
1. Node 17+ is throwing an internal OpenSSL error when establishing a secure TLS connection to the iOS device. This is caused by the explicit `secureProtocol` option, which is set to `TLSv1_method`.
2. We don't have any error event handlers when opening (secure) sockets to iOS devices.

These two issues together result in `expo run:ios` not-installing the app on the device. Luckily, it does build the app. But, due to the missing error handlers, we are swallowing the underlying error and the installation part is [stuck on a promise](https://github.com/expo/expo/blob/main/packages/%40expo/cli/src/run/ios/appleDevice/AppleDevice.ts#L143) that will never be resolved.

Here is the chain that's getting blocked by issue 2 (caused by issue 1)
- `clientManager.getMobileImageMounterClient`
- `this.getServiceClient('com.apple.mobile.mobile_image_mounter', MobileImageMounterClient)`
- `lockdowndClient.startService(name)` is invoked (from `MobileImageMounterClient`)
- `LockdownProtocolClient.sendMessage` is invoked (with `StartService`, `com.apple.mobile.mobile_image_mounter`)
- `ProtocolClient.sendMessage` is invoked
  - Socket TLS handshake fails, but the `socket.on('error')` is not set. So the error is never received.

# How

- Removed the explicit `TLSv1_method` from `tls.createSecureContext`
- Added error handlers to surface underlying errors

After removing the fixed TLS method, it seems to use TLSv1.2 by default.

# Test Plan

Test `main` on Node 17+, while adding `enableTrace: true` and `socket.on('error', (error) => console.log(error));` to the changes I made
> This should fail with an internal socket/openssl error

Test this PR on Node 16, and Node 18
> This should NOT fail

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/@expo/cli/src/run/ios/appleDevice/client/LockdowndClient.ts [hunk 1]
```diff
@@ -128,7 +128,6 @@ export class LockdowndClient extends ServiceClient<LockdownProtocolClient> {
       if (resp.EnableSessionSSL) {
         this.protocolClient.socket = new tls.TLSSocket(this.protocolClient.socket, {
           secureContext: tls.createSecureContext({
-            secureProtocol: 'TLSv1_method',
```
Comment (EvanBacon) [inline packages/@expo/cli/src/run/ios/appleDevice/client/LockdowndClient.ts:131]: Add a comment note explaining why we can't add this property back.

**Review by EvanBacon — APPROVED** [review]
Lgtm

File: packages/@expo/cli/src/run/ios/appleDevice/client/LockdowndClient.ts (hunk 1 above)
Comment (byCedric) [reply packages/@expo/cli/src/run/ios/appleDevice/client/LockdowndClient.ts:131]: Done, hope it makes sense.

TL;DR; `secureProtocol` is deprecated in favor of `minVersion` and `maxVersion`, both with proper default values to establish a secure connection based on what both Node and the device support.
